packages/jest-cli/src/reporters/CoverageReporter.js

```diff
@@ -35,27 +39,28 @@ class CoverageReporter extends BaseReporter {
     aggregatedResults: AggregatedResult,
   ) {
     if (testResult.coverage) {
-      const testFilePath = testResult.testFilePath;
-      this._collector.add(testResult.coverage);
-      if (!this._testCollectors[testFilePath]) {
-        this._testCollectors[testFilePath] = new istanbul.Collector();
-      }
-      this._testCollectors[testFilePath].add(testResult.coverage);
+      this._coverageMap.merge(testResult.coverage);
     }
   }
 
   onRunComplete(config: Config, aggregatedResults: AggregatedResult) {
-    const reporter = new istanbul.Reporter();
+    const reporter = createReporter();
     try {
       if (config.coverageDirectory) {
         reporter.dir = config.coverageDirectory;
       }
-      reporter.addAll(config.coverageReporters);
-      reporter.write(this._collector, true, () => {});
-    } catch (e) {}
+      reporter.addAll(config.coverageReporters || []);
+      reporter.write(this._coverageMap);
+    } catch (e) {
+      throw new Error(`
+        Failed to write coverage reports:
```

Comment:
I think in the past we decided not to throw here so that people won't be slowed down. Can we turn this into a `console.error(chalk.red())` instead? It will give people signal that we couldn't write the report but it won't stop them from running Jest.